pandas/tests/reshape/test_pivot.py

```diff
@@ -1325,6 +1322,73 @@ def test_crosstab_normalize(self):
                                 margins=True)
         tm.assert_frame_equal(test_case, norm_sum)
 
+    def test_crosstab_norm_margins_with_multiindex(self):
+        # GH 15150
+        a = np.array(['foo', 'bar', 'foo', 'bar', 'bar', 'foo'])
+        b = np.array(['one', 'one', 'two', 'one', 'two', 'two'])
```

Comment:
this test is very hard to read, make it in several sections

input data
expected
result
assert_frame_equal

then repeat for cases ....